### ISSUE TITLE
Entering statement starting in 'w' in debugger freezes following code execution

### ISSUE THREAD

**Issue by flixha** [spyder-ide/spyder#11537]
### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [x] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [x] Tried basic troubleshooting (if a bug/error)
    * [x] Restarted Spyder (this solves the freeze)
    * [x] Reset preferences with ``spyder --reset``
    * [x] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [x] Tried the other applicable steps from the Troubleshooting Guide
* [x] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description
After entering a statement starting in "w" in the debugger, Spyder stops executing any code (through "run", or in python terminal). "w" alone in the debugger console prints a trace stack, so entering a statement following "w" (e.g., w = a) will show an expected error (see below in traceback). Following that error, spyder does not execute any code any more (neither via "run" or in the python console). A kernel restart doesn't help either, only rebooting spyder will fix the issue.

### What steps reproduce the problem?
1. Enter the following code in a script:
```
a = 1+1
w = a
```
2. Set a breakpoint at line 2:
3. Start the debugger
4. Enter "w = a" into debugger console.
5. Try to run the script, or try to run `1+1` in the python console.

### What is the expected output? What do you see instead?
Entering "w" in the debugger prints a stack trace, so entering a statement following "w" indeed should return an error. But after the error occurs, the user should be able to interact with spyder again without restarting spyder.

### Paste Traceback/Error Below (if applicable)

(this error precedes the actual problem - the freeze)
```Traceback (most recent call last):

  File "/Volumes/MacHD3_CaseSensitive/Documents2/ArrayWork/Python/testSpyderReactivity.py", line 10, in <module>
    c = a

  File "/Volumes/MacHD3_CaseSensitive/Documents2/ArrayWork/Python/testSpyderReactivity.py", line 10, in <module>
    c = a

  File "/Volumes/MacHD1/Users/felix/Software/miniconda/miniconda3/envs/TestEnv/lib/python3.7/bdb.py", line 88, in trace_dispatch
    return self.dispatch_line(frame)

  File "/Volumes/MacHD1/Users/felix/Software/miniconda/miniconda3/envs/TestEnv/lib/python3.7/bdb.py", line 112, in dispatch_line
    self.user_line(frame)

  File "/Volumes/MacHD1/Users/felix/Software/miniconda/miniconda3/envs/TestEnv/lib/python3.7/pdb.py", line 259, in user_line
    self.interaction(frame, None)

  File "/Volumes/MacHD1/Users/felix/Software/miniconda/miniconda3/envs/TestEnv/lib/python3.7/site-packages/spyder_kernels/customize/spyderpdb.py", line 143, in interaction
    self._cmdloop()

  File "/Volumes/MacHD1/Users/felix/Software/miniconda/miniconda3/envs/TestEnv/lib/python3.7/site-packages/spyder_kernels/customize/spyderpdb.py", line 302, in _cmdloop
    self.cmdloop()

  File "/Volumes/MacHD1/Users/felix/Software/miniconda/miniconda3/envs/TestEnv/lib/python3.7/cmd.py", line 138, in cmdloop
    stop = self.onecmd(line)

  File "/Volumes/MacHD1/Users/felix/Software/miniconda/miniconda3/envs/TestEnv/lib/python3.7/pdb.py", line 416, in onecmd
    return cmd.Cmd.onecmd(self, line)

  File "/Volumes/MacHD1/Users/felix/Software/miniconda/miniconda3/envs/TestEnv/lib/python3.7/cmd.py", line 217, in onecmd
    return func(arg)

  File "/Volumes/MacHD1/Users/felix/Software/miniconda/miniconda3/envs/TestEnv/lib/python3.7/site-packages/IPython/core/debugger.py", line 625, in do_where
    context = int(arg)

ValueError: invalid literal for int() with base 10: '=a'

```

## Versions

* Spyder version: 4.0.1 (and tested in master-version)
* Python version: 3.7.3 64-bit
* Qt version: Qt 5.12.5
* PyQt version: PyQt5 5.12.3 
* Operating System name/version: Darwin 17.7.0 

### Dependencies

```
applaunchservices >=0.1.7    :  0.2.1 (OK)
atomicwrites >=1.2.0         :  1.3.0 (OK)
chardet >=2.0.0              :  3.0.4 (OK)
cloudpickle >=0.5.0          :  1.2.2 (OK)
diff_match_patch >=20181111  :  20181111 (OK)
intervaltree                 :  None (OK)
IPython >=4.0                :  7.12.0 (OK)
jedi =0.14.1                 :  0.14.1 (OK)
nbconvert >=4.0              :  5.6.1 (OK)
numpydoc >=0.6.0             :  0.9.2 (OK)
pexpect >=4.4.0              :  4.8.0 (OK)
pickleshare >=0.4            :  0.7.5 (OK)
psutil >=0.3                 :  5.6.7 (OK)
pygments >=2.0               :  2.5.2 (OK)
pylint >=0.25                :  2.4.4 (OK)
pyls >=0.31.2;<0.32.0        :  0.31.7 (OK)
zmq >=17                     :  18.1.1 (OK)
qdarkstyle >=2.7             :  2.8 (OK)
qtawesome >=0.5.7            :  0.6.1 (OK)
qtconsole >=4.6.0            :  4.6.0 (OK)
qtpy >=1.5.0                 :  1.9.0 (OK)
rtree >=0.8.3                :  0.9.4 (OK)
sphinx >=0.6.6               :  2.4.0 (OK)
spyder_kernels >=1.8.1;<2.0.0:  1.8.1 (OK)
watchdog                     :  None (OK)
cython >=0.21                :  None (NOK)
matplotlib >=2.0.0           :  None (NOK)
numpy >=1.7                  :  None (NOK)
pandas >=0.13.1              :  None (NOK)
scipy >=0.17.0               :  None (NOK)
sympy >=0.7.3                :  None (NOK)
```

**Comment by ccordoba12** [comment]
The problem is that you're redefining a core Pdb command (yes, `w` is command), so there's no way to recover from this error, sorry.

The only thing we can do is to redefine Pdb commands to magics to avoid this kind of problems. But that's planned for Spyder 5.

**Comment by flixha** [comment]
Thanks a lot for the quick reply and it's great to hear about such plans for Spyer 5! I know that what I'm entering in that situation is not smart as `w` is already a command in pdb. It happens because I forget myself and I have too many variables tarting in "w" ;-). 
The thing is that none of the other pdb commands `b`, `n`, `b`, `s` etc. lead to this situation where the user has to restart spyder to continue with any kind of work (because spyder does not react to any commands afterwards, even after a kernel restart). That's why it appeared like a bug.

**Comment by ccordoba12** [comment]
I'm sorry to hear that but this is a Pdb limitation that we can't address in a maintenance version (i.e. in Spyder 4).

**Comment by impact27** [comment]
Fixed in https://github.com/ipython/ipython/pull/12219

**Comment by ccordoba12** [comment]
Thanks @impact27 for your help with this one!

@flixha, the fix is now part of IPython 7.14.0, released yesterday. Conda packages for that version should be available next week.
